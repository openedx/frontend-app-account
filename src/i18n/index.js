// Placeholder be overridden by `make pull_translations`
export default {
  ar: {},
  'zh-hk': {},
  'zh-cn': {},
  uk: {},
  'tr-tr': {},
  th: {},
  te: {},
  ru: {},
  'pt-pt': {},
  'pt-br': {},
  'it-it': {},
  id: {},
  hi: {},
  he: {},
  'fr-ca': {},
  fa: {},
  'es-es': {},
  'es-419': {},
  el: {},
  'de-de': {},
  da: {},
  bo: {},
};
