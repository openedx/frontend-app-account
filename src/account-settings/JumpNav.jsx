import { getConfig } from '@edx/frontend-platform';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { breakpoints, useWindowSize } from '@openedx/paragon';
import classNames from 'classnames';
import React from 'react';
import { NavHashLink } from 'react-router-hash-link';
import Scrollspy from 'react-scrollspy';
import messages from './AccountSettingsPage.messages';

const JumpNav = ({
  intl,
}) => {
  const stickToTop = useWindowSize().width > breakpoints.small.minWidth;

  return (
    <div className={classNames('jump-nav px-2.25', { 'jump-nav-sm position-sticky pt-3': stickToTop })}>
      <Scrollspy
        items={[
          'basic-information',
          'profile-information',
          'social-media',
          'notifications',
          'site-preferences',
          'linked-accounts',
          'delete-account',
        ]}
        className="list-unstyled"
        currentClassName="font-weight-bold"
      >
        <li>
          <NavHashLink to="#basic-information">
            {intl.formatMessage(messages['account.settings.section.account.information'])}
          </NavHashLink>
        </li>
        <li>
          <NavHashLink to="#profile-information">
            {intl.formatMessage(messages['account.settings.section.profile.information'])}
          </NavHashLink>
        </li>
        <li>
          <NavHashLink to="#social-media">
            {intl.formatMessage(messages['account.settings.section.social.media'])}
          </NavHashLink>
        </li>
        <li>
          <NavHashLink to="#notifications">
            {intl.formatMessage(messages['notification.preferences.notifications.label'])}
          </NavHashLink>
        </li>
        <li>
          <NavHashLink to="#site-preferences">
            {intl.formatMessage(messages['account.settings.section.site.preferences'])}
          </NavHashLink>
        </li>
        <li>
          <NavHashLink to="#linked-accounts">
            {intl.formatMessage(messages['account.settings.section.linked.accounts'])}
          </NavHashLink>
        </li>
        {getConfig().ENABLE_ACCOUNT_DELETION
          && (
          <li>
            <NavHashLink to="#delete-account">
              {intl.formatMessage(messages['account.settings.jump.nav.delete.account'])}
            </NavHashLink>
          </li>
          )}
      </Scrollspy>
    </div>
  );
};

JumpNav.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(JumpNav);
