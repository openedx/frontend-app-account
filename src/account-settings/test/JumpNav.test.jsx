import React from 'react';
import { render, screen } from '@testing-library/react';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
import { AppProvider } from '@edx/frontend-platform/react';
import { initializeMockApp, mergeConfig, setConfig } from '@edx/frontend-platform';

import JumpNav from '../JumpNav';
import configureStore from '../../data/configureStore';

const IntlJumpNav = injectIntl(JumpNav);

describe('JumpNav', () => {
  mergeConfig({
    ENABLE_ACCOUNT_DELETION: true,
  });

  let props = {};
  let store;

  beforeEach(() => {
    initializeMockApp({
      authenticatedUser: {
        userId: 3,
        username: 'abc123',
        administrator: true,
        roles: [],
      },
    });

    props = {
      intl: {},
    };
    store = configureStore({
      notificationPreferences: {
        showPreferences: false,
      },
    });
  });

  it('should not render delete account link', async () => {
    setConfig({
      ENABLE_ACCOUNT_DELETION: false,
    });

    render(
      <IntlProvider locale="en">
        <AppProvider store={store}>
          <IntlJumpNav {...props} />
        </AppProvider>
      </IntlProvider>,
    );

    expect(await screen.queryByText('Delete My Account')).toBeNull();
  });

  it('should render delete account link', async () => {
    setConfig({
      ENABLE_ACCOUNT_DELETION: true,
    });

    props = {
      ...props,
    };

    render(
      <IntlProvider locale="en">
        <AppProvider store={store}>
          <IntlJumpNav {...props} />
        </AppProvider>
      </IntlProvider>,
    );

    expect(await screen.findByText('Delete My Account')).toBeVisible();
  });
});
