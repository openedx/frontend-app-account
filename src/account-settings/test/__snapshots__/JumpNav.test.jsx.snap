// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JumpNav should not render Optional Information or delete account link 1`] = `
<div
  data-testid="redux-provider"
>
  <div
    data-testid="browser-router"
  >
    <div
      className="jump-nav px-2.25 jump-nav-sm position-sticky pt-3"
    >
      <ul
        className="list-unstyled"
        style={{}}
      >
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#basic-information"
            isActive={[Function]}
            onClick={[Function]}
          >
            Account Information
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#profile-information"
            isActive={[Function]}
            onClick={[Function]}
          >
            Profile Information
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#social-media"
            isActive={[Function]}
            onClick={[Function]}
          >
            Social Media Links
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#notifications"
            isActive={[Function]}
            onClick={[Function]}
          >
            Notifications
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#site-preferences"
            isActive={[Function]}
            onClick={[Function]}
          >
            Site Preferences
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#linked-accounts"
            isActive={[Function]}
            onClick={[Function]}
          >
            Linked Accounts
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`JumpNav should render Optional Information and delete account link 1`] = `
<div
  data-testid="redux-provider"
>
  <div
    data-testid="browser-router"
  >
    <div
      className="jump-nav px-2.25 jump-nav-sm position-sticky pt-3"
    >
      <ul
        className="list-unstyled"
        style={{}}
      >
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#basic-information"
            isActive={[Function]}
            onClick={[Function]}
          >
            Account Information
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#profile-information"
            isActive={[Function]}
            onClick={[Function]}
          >
            Profile Information
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#social-media"
            isActive={[Function]}
            onClick={[Function]}
          >
            Social Media Links
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#notifications"
            isActive={[Function]}
            onClick={[Function]}
          >
            Notifications
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#site-preferences"
            isActive={[Function]}
            onClick={[Function]}
          >
            Site Preferences
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#linked-accounts"
            isActive={[Function]}
            onClick={[Function]}
          >
            Linked Accounts
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-current="page"
            className="active"
            href="/#delete-account"
            isActive={[Function]}
            onClick={[Function]}
          >
            Delete My Account
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
